nmdc_runtime/site/translation/neon_benthic_translator.py

```diff
@@ -0,0 +1,550 @@
+    "stream": {"term_id": "ENVO:01000253", "term_name": "freshwater river biome"}
+}
+
+BENTHIC_LOCAL_SCALE_MAPPINGS = {
+    "pool": {"term_id": "ENVO:03600094", "term_name": "stream pool"},
+    "run": {"term_id": "ENVO:03600095", "term_name": "stream run"},
+    "step pool": {"term_id": "ENVO:03600096", "term_name": "step pool"},
+    "riffle": {"term_id": "ENVO:00000148", "term_name": "riffle"},
+    "stepPool": {"term_id": "ENVO:03600096", "term_name": "step pool"},
+}
+
+BENTHIC_ENV_MEDIUM_MAPPINGS = {
+    "plant-associated": {
+        "term_id": "ENVO:01001057",
+        "term_name": "environment associated with a plant part or small plant",
+    },
+    "sediment": {"term_id": "ENVO:00002007", "term_name": "sediment"},
+    "biofilm": {"term_id": "ENVO:00002034", "term_name": "biofilm"},
+}
+
+
+class NeonBenthicDataTranslator(Translator):
+    def __init__(self, benthic_data: dict, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+
+        self.conn = sqlite3.connect("neon.db")
+        requests_cache.install_cache("neon_api_cache")
+
+        neon_amb_data_tables = (
+            "mms_benthicMetagenomeSequencing",
+            "mms_benthicMetagenomeDnaExtraction",
+            "amb_fieldParent",
+        )
+
+        if all(k in benthic_data for k in neon_amb_data_tables):
+            benthic_data["mms_benthicMetagenomeSequencing"].to_sql(
+                "mms_benthicMetagenomeSequencing",
+                self.conn,
+                if_exists="replace",
+                index=False,
+            )
+            benthic_data["mms_benthicMetagenomeDnaExtraction"].to_sql(
+                "mms_benthicMetagenomeDnaExtraction",
+                self.conn,
+                if_exists="replace",
+                index=False,
+            )
+            benthic_data["amb_fieldParent"].to_sql(
+                "amb_fieldParent", self.conn, if_exists="replace", index=False
+            )
+        else:
+            raise ValueError(
+                f"You are missing one of the aquatic benthic microbiome tables: {neon_amb_data_tables}"
+            )
+
+        neon_envo_mappings_file = "https://raw.githubusercontent.com/microbiomedata/nmdc-schema/main/assets/neon_mixs_env_triad_mappings/neon-nlcd-local-broad-mappings.tsv"
+        neon_envo_terms = pd.read_csv(neon_envo_mappings_file, delimiter="\t")
+        neon_envo_terms.to_sql(
+            "neonEnvoTerms", self.conn, if_exists="replace", index=False
+        )
+
+        neon_raw_data_file_mappings_file = "https://raw.githubusercontent.com/microbiomedata/nmdc-schema/main/assets/misc/neon_raw_data_file_mappings.tsv"
+        self.neon_raw_data_file_mappings_df = pd.read_csv(
+            neon_raw_data_file_mappings_file, delimiter="\t"
+        )
+        self.neon_raw_data_file_mappings_df.to_sql(
+            "neonRawDataFile", self.conn, if_exists="replace", index=False
+        )
+
+    def get_site_by_code(self, site_code: str) -> str:
```

Comment:
Similar to above comment it would be great to see this implemented as a separate `@op` that _just_ produces a mapping from site code to location string (perhaps by calling the `https://data.neonscience.org/api/v0/sites` endpoint) and handing that mapping to the `Translator` class.